osbenchmark/metrics.py

```diff
@@ -1665,30 +1675,52 @@ def encode_float_key(k):
     return str(float(k)).replace(".", "_")
 
 
-def percentiles_for_sample_size(sample_size):
-    # if needed we can come up with something smarter but it'll do for now
+def filter_percentiles_by_sample_size(sample_size, percentiles):
+    # Don't show percentiles if there aren't enough samples for the value to be distinct.
+    # For example, we should only show p99.9, p45.6, or p0.01 if there are at least 1000 values.
+    # If nothing is suitable, default to just returning [100] rather than an empty list.
     if sample_size < 1:
         raise AssertionError("Percentiles require at least one sample")
-    elif sample_size == 1:
-        return [100]
-    elif 1 < sample_size < 10:
-        return [50, 100]
-    elif 10 <= sample_size < 100:
-        return [50, 90, 100]
-    elif 100 <= sample_size < 1000:
-        return [50, 90, 99, 100]
-    elif 1000 <= sample_size < 10000:
-        return [50, 90, 99, 99.9, 100]
+
+    filtered_percentiles = []
+    # Treat the 1 and 2-10 case separately, to return p50 and p100 if present
+    if sample_size == 1:
+        filtered_percentiles = [100]
+    elif sample_size < 10:
+        for p in [50, 100]:
+            if p in percentiles:
+                filtered_percentiles.append(p)
     else:
-        return [50, 90, 99, 99.9, 99.99, 100]
+        effective_sample_size = 10 ** (int(math.log10(sample_size))) # round down to nearest power of ten
+        delta = 0.000001 # If (p / 100) * effective_sample_size is within this value of a whole number,
+        # assume the discrepancy is due to floating point and allow it
+        filtered_percentiles = []
+        for p in percentiles:
+            fraction = p / 100
+
+            # check if fraction * effective_sample_size is close enough to a whole number
+            if abs((effective_sample_size * fraction) - round(effective_sample_size*fraction)) < delta:
+                filtered_percentiles.append(p)
+    # if no percentiles are suitable, just return 100
+    if len(filtered_percentiles) == 0:
+        return [100]
+    return filtered_percentiles
 
+def percentiles_for_sample_size(sample_size, latency_percentiles=None):
+    # If latency_percentiles is present, as a list, also display those values (assuming there are enough samples)
+    percentiles = [50, 90, 99, 99.9, 99.99, 100]
```

Comment:
Might be better to make this a constant that's referenced and instead of initializing it here